✨ [FEAT] ai-service 모듈 생성

ByeongminJeong opened the pull request:
## 🔘Part

- [x] Gemini API 연동, 상품/음식점 설명 생성
- [x] OpenWeatherMap API + Gemini API 기반 날씨별 메뉴 추천
- [x] AI 로그 저장 구현

  <br/>
## ➕ 이슈 링크

- #13 

<br/>

## 🔎 작업 내용

- Gemini API 연동
메뉴 설명 생성: 메뉴명 → AI가 매력적인 설명 생성
가게 설명 생성: 가게 정보 → AI가 소개글, 환영메시지, 추천메뉴 생성

- 날씨 기반 메뉴 추천(OpenWeatherMap + Gemini AI)
사용자 주소 조회 → 실시간 날씨 API → AI 메뉴 추천
(온도, 습도, 날씨상태 데이터를 불러옴 → 맞춤 메뉴 + 가게 + 조언 생성)

- AI 로그 저장
question (AI에게 보낸 프롬프트)
result (AI 응답)
requestType (PRODUCT/RESTAURANT/WEATHER_MENU)
응답 성공/실패 여부

  <br/>

## 이미지 첨부

<img width="1493" height="536" alt="image" src="https://github.com/user-attachments/assets/4397da0d-3f8e-4ca7-8aa9-5d0c7019d575" />


<img width="1593" height="550" alt="image" src="https://github.com/user-attachments/assets/6b453b73-7ce7-49a4-938b-86b6caac8435" />


- 메뉴 설명 생성
<img width="1126" height="589" alt="image" src="https://github.com/user-attachments/assets/cf9ff5ca-da49-4395-8768-e00401be52b3" />


- 가게 설명 생성
<img width="1129" height="753" alt="image" src="https://github.com/user-attachments/assets/df127a8f-b96a-40b6-9e1c-142347ab71d7" />


- 날씨에 따른 메뉴 추천
<img width="1126" height="691" alt="image" src="https://github.com/user-attachments/assets/31be9294-e532-4721-937d-bd1fe23190d8" />



<br/>

ByeongminJeong committed:
✨ feat: ai-service 모듈 생성

- Gemini API 연동, 상품/음식점 설명 생성
- OpenWeatherMap API 기반 날씨별 메뉴 추천
- AI 로그 저장 구현

shiien14 commented on ai-service/settings.gradle:
settings.gradle 삭제해주세요 !!

shiien14 commented on ai-service/src/main/java/com/example/cloudfour/aiservice/AiServiceApplication.java:
EurekaClient 어노테이션 추가해주세요

ByeongminJeong committed:
🐛 fix: settings.gradle 삭제

ByeongminJeong committed:
🐛 fix: AI 로그 저장 시 question/result 컬럼에 전체 JSON 문자열 저장

- 모든 서비스(날씨 추천, 메뉴 설명, 가게 설명)에서 saveAiLog 호출 시 result 파라미터에 전체 JSON 문자열을 저장하도록 통일
- import .*; 수정

jh010303 commented on ai-service/build.gradle:
```diff
@@ -0,0 +1,53 @@
+
+tasks.named('test') {
+    useJUnitPlatform()
+}
```
tasks.register("prepareKotlinBuildScriptModel"){} 추가해주세요

jh010303 commented on ai-service/src/main/java/com/example/cloudfour/aiservice/dto/GeminiRequestDTO.java:
```diff
@@ -0,0 +1,31 @@
+
+@Data
+@Builder
+@NoArgsConstructor
```
dto에서는 @Getter, @Builder만 사용해주세요

jh010303 commented on ai-service/src/main/java/com/example/cloudfour/aiservice/repository/AiLogRepository.java:
```diff
@@ -0,0 +1,16 @@
+import java.util.List;
+import java.util.UUID;
+
+@Repository
```
@Repository 제거해주세요

jh010303 commented on ai-service/src/main/java/com/example/cloudfour/aiservice/service/command/ProductDescriptionCommandServiceImpl.java:
```diff
@@ -0,0 +1,213 @@
+        String keyFeatures = "";
+        String suggestedTags = "";
+        
+        // 여러 줄로 분리하여 각 줄을 확인
```
주석 제거해주세요

jh010303 commented on ai-service/src/main/java/com/example/cloudfour/aiservice/service/command/ProductDescriptionCommandServiceImpl.java:
```diff
@@ -0,0 +1,213 @@
+        }
+        
+        // 만약 파싱이 실패했다면 전체 텍스트에서 키워드로 찾기
+        if (description.isEmpty() && responseText.contains("메뉴 설명")) {
```
if~ elseif 문 너무 길어서 따로 클래스로 뺄 수 있으면 빼주세요

ByeongminJeong committed:
🔨 refactor: AI 서비스 코드 구조 개선 및 한글 도시명 지원 추가

- Lombok 어노테이션 제거, @Data 제거, @Getter @Builder만 사용하도록 함
- Builder, Parser, Converter로 코드 모듈화
- 한글 도시명을 영어로 변환하는 CityTranslator 추가
- WeatherConstants로 날씨 코드 관리
- 불필요한 설정 파일 제거 (SwaggerConfig, JpaConfig)
- 긴 서비스 로직을 별도 클래스로 분리(if -else 문)

ByeongminJeong committed:
🔨 refactor: application.yml 수정

jh010303 reviewed (approved):
고생하셨습니다 !! 😄
